pod/annotations.go

```diff
@@ -167,6 +167,19 @@ const (
 	// A full example would be "capabilities.containers.netflix.com/mycoolsidecar: FUSE,Default"
 	// Note that arbitrary combinations of capabilities are not necissarily valid.
 	AnnotationKeySuffixContainersCapabilities = "capabilities"
+	// AnnotationsKeySuffixContainersStartBefore provides a comma-separated
+	// list of other containers in the pod that should only be started
+	// *after* the specified container starts up. If startup order isn't
+	// specified, sidecars may be started in any order. Conflicting startup
+	// order requests are treated as an error.
+	AnnotationKeySuffixContainersStartBefore = "start-before"
+	// AnnotationsKeySuffixContainersStartBefore provides a comma-separated
+	// list of other containers in the pod that must be started *before* the
+	// specified container starts up. If startup order isn't specified,
+	// sidecars may be started in any order. Conflicting startup order
+	// requests are treated as an error.
+	AnnotationKeySuffixContainersStartAfter = "start-after"
```

Comment:
Can you talk a little bit about the use cases here? I'm worried that `start-before` and `start-after` won't be enough. For example, if a service needs to `bind()` to a socket before another service, starting the service that needs to `bind()` first doesn't necessarily guarantee it `bind()`s before the second service needs it.

systemd solves this with socket activation, which is a reasonably complex hammer. I'm just curious if we need something like that (or similar ordering for supplied storage or whatever).

Reply:
We have some use cases in hybrid development where this is important. We need to sync config files up to the remote pod, but the sidecar that handles file sync needs to start up and get those files in place before other sidecars start.

What I had in mind for this is that when container A needs to start before container B, container B shouldn't be started until container A's health check has passed. The health checks should set up accordingly (i.e. it shouldn't pass until all necessary listening sockets are bound).

Reply:
Ok, the health checks bit is what I was missing, thanks. Can you send another patch to add that language to the docstring?

Reply:
Sure, here's a PR: #59